### ISSUE TITLE
Crash on iOS when calling await OktaContext.Current.SignOutAsync(); System.ObjectDisposedException

### ISSUE THREAD

**Issue by kensykora** [okta/okta-oidc-xamarin#71]
When trying to call `await OktaContext.Current.SignOutAsync();` the library throws an exception, causing a crash. I have tried a few ways to work around it:

 - Calling `SignOutAsync()` on the main thread -- made no impact
 - Reinstantiating `OktaContext.Current = new OktaContext()` before trying to call `SignOutAsync()` -- this didn't work, threw a different error 

This issue does not occur on Android

## Current behavior
When calling `SignOutAsync() on iOS, the following exception is thrown and stack trace produced:

```
2021-11-03 14:48:53.179823-0500 Courier.iOS[21063:883270] [AppCenterCrashes] ERROR: +[MSACWrapperLogger MSACWrapperLog:tag:level:]/10 Unhandled Exception:
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'PlatformRenderer'.
  at Foundation.NSObject.get_SuperHandle () [0x00012] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSObject2.cs:541 
  at UIKit.UIViewController.PresentViewController (UIKit.UIViewController viewControllerToPresent, System.Boolean animated, System.Action completionHandler) [0x00059] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/build/ios/native/UIKit/UIViewController.g.cs:713 
  at UIKit.UIViewController.PresentViewControllerAsync (UIKit.UIViewController viewControllerToPresent, System.Boolean animated) [0x00011] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/build/ios/native/UIKit/UIViewController.g.cs:722 
  at Okta.Xamarin.iOS.iOsOidcClient.LaunchBrowser (System.String url) [0x00018] in <1e8570cde6084814bc9452b5bad8bb1b>:0 
  at Okta.Xamarin.OidcClient.SignOutOfOktaAsync (Okta.Xamarin.IOktaStateManager stateManager) [0x0008d] in <673d6a28a6134bd5ac2960a0b16cb008>:0 
  at Okta.Xamarin.OktaContext.SignOutAsync (Okta.Xamarin.IOidcClient oidcClient) [0x0004e] in <673d6a28a6134bd5ac2960a0b16cb008>:0 
  at Courier.Services.OktaAuthService.Logout () [0x0002b] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/Services/OktaAuthService.cs:72 
  at Courier.ViewModels.MainViewModel.DoLogout () [0x00024] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/ViewModels/MainViewModel.cs:87 
  at Courier.ViewModels.MainViewModel.<get_Logout>b__17_0 () [0x0001f] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/ViewModels/MainViewModel.cs:41 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:178 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00013] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:69 
  at Courier.iOS.Application.Main (System.String[] args) [0x00001] in /Users/kensykora/Workspace/app-courier-xamarin/Courier.iOS/Main.cs:17
2021-11-03 14:48:53.572697-0500 Courier.iOS[21063:883270] Unhandled managed exception: Cannot access a disposed object.
Object name: 'PlatformRenderer'. (System.ObjectDisposedException)
  at Foundation.NSObject.get_SuperHandle () [0x00012] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSObject2.cs:541 
  at UIKit.UIViewController.PresentViewController (UIKit.UIViewController viewControllerToPresent, System.Boolean animated, System.Action completionHandler) [0x00059] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/build/ios/native/UIKit/UIViewController.g.cs:713 
  at UIKit.UIViewController.PresentViewControllerAsync (UIKit.UIViewController viewControllerToPresent, System.Boolean animated) [0x00011] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/build/ios/native/UIKit/UIViewController.g.cs:722 
  at Okta.Xamarin.iOS.iOsOidcClient.LaunchBrowser (System.String url) [0x00018] in <1e8570cde6084814bc9452b5bad8bb1b>:0 
  at Okta.Xamarin.OidcClient.SignOutOfOktaAsync (Okta.Xamarin.IOktaStateManager stateManager) [0x0008d] in <673d6a28a6134bd5ac2960a0b16cb008>:0 
  at Okta.Xamarin.OktaContext.SignOutAsync (Okta.Xamarin.IOidcClient oidcClient) [0x0004e] in <673d6a28a6134bd5ac2960a0b16cb008>:0 
  at Courier.Services.OktaAuthService.Logout () [0x0002b] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/Services/OktaAuthService.cs:72 
  at Courier.ViewModels.MainViewModel.DoLogout () [0x00024] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/ViewModels/MainViewModel.cs:87 
  at Courier.ViewModels.MainViewModel.<get_Logout>b__17_0 () [0x0001f] in /Users/kensykora/Workspace/app-courier-xamarin/Courier/ViewModels/MainViewModel.cs:41 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:178 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00013] in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:69 
  at Courier.iOS.Application.Main (System.String[] args) [0x00001] in /Users/kensykora/Workspace/app-courier-xamarin/Courier.iOS/Main.cs:17
```
## Expected behavior

Sign out webview should be presented and the appropriate events raised

## Minimal reproduction of the problem with instructions

Can provide if needed.

## Environment

Nuget Packages:
 - Xamarin Forms 5.0.0.2196
 - Okta.Xamarin 3.0.0

macOS Big Sur 11.4, Mac Mini, M1 2020
XCode 13.1

Experienced on iOS SImulator iPhone 13 running iOS 15.0

Build Environment

```
Visual Studio Community 2019 for Mac
Version 8.10.12 (build 16)
Installation UUID: d226ec8f-b62a-43bd-b2f0-ee1f4f6a79a5
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

Mono Framework MDK
Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

Roslyn (Language Service)
3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

NuGet
Version: 5.9.0.7134

.NET Core SDK
SDK: /usr/local/share/dotnet/sdk/5.0.402/Sdks
SDK Versions:
	5.0.402
	3.1.414
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

.NET Core Runtime
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.11
	3.1.20

.NET Core 3.1 SDK
SDK: 3.1.414

Xamarin.Profiler
Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode 13.1 (19466)
Build 13A1030d

Xamarin Designer
Version: 16.11.0.39
Hash: cd672761d
Branch: remotes/origin/d16-11
Build date: 2021-10-06 20:09:18 UTC

Xamarin.Mac
Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

Xamarin.iOS
Version: 15.2.0.1 (Visual Studio Community)
Hash: 8fd9e6289
Branch: xcode13.1
Build date: 2021-10-26 10:13:56-0400

Xamarin.Android
Version: 12.0.0.3 (Visual Studio Community)
Commit: xamarin-android/d16-11/f0e3c2d
Android SDK: /Users/kensykora/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/d16-11@476bb5b
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-11@87af37b

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home
1.8.0_302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:12 UTC

Android Device Manager
Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-09-28 20:47:30 UTC

Build Information
Release ID: 810120016
Git revision: e4f3f43ac50d8e0f83a71dbcb666bb1611a661d7
Build date: 2021-10-28 12:10:44-04
Build branch: release-8.10

Operating System
Mac OS X 10.16.0
Darwin 20.5.0 Darwin Kernel Version 20.5.0
    Sat May  8 05:10:31 PDT 2021
    root:xnu-7195.121.3~9/RELEASE_ARM64_T8101 x86_64
```

**Comment by andriizhegurov-okta** [comment]
Hi @kensykora, thank you for reporting this issue. I'll add an item to our backlog to investigate this.

**Comment by bryanapellanes-okta** [comment]
@kensykora,
Thanks again for reporting this issue!  To help my investigation, may I have you provide a Visual Studio solution that reproduces the issue?

**Comment by bryanapellanes-okta** [comment]
@kensykora,
I believe your PR #72 should correct this issue.  Please feel free to reopen if I am incorrect.  I will publish a bug fix release after #78 is reviewed and merged.
